src/stream/src/executor/source/source_executor.rs

```diff
@@ -530,8 +536,17 @@ impl<S: StateStore> SourceExecutor<S> {
 
                     let epoch = barrier.epoch;
 
+                    if self.is_shared
+                        && is_uninitialized
+                        && barrier.has_new_downstream(self.actor_ctx.id)
+                    {
+                        stream.resume_stream();
```

Comment:
Shall we also handle the `Stop` mutation, i.e., pause the source again if all downstream jobs have been dropped? This appears to also apply to the optimization of on-conflict handling (even more reasonable).

Reply:
NTFS